1.5/Source/PawnStorages/PawnStorages/WorkGiver_Warden_TakeToStorage.cs

```diff
@@ -34,19 +34,36 @@ private static Job TakeToPreferredStorageJob(Pawn prisoner, Pawn warden)
 		Job job = JobMaker.MakeJob(PS_DefOf.PS_CaptureInPawnStorage, prisoner, storage);
 		job.count = 1;
 		return job;
-	}
+    }
 
-	public static ThingWithComps GetStorageForPawn(Pawn prisoner, bool assign = false)
-	{
-		ThingWithComps existingAssigned = CompAssignableToPawn_PawnStorage.compAssiblables.FirstOrDefault(c => c.assignedPawns.Contains(prisoner))?.parent;
-		if (existingAssigned != null)
-		{
-			return existingAssigned;
-		}
+    public static ThingWithComps GetStorageForPawn(Pawn prisoner, bool assign = false)
+    {
+        ThingWithComps existingAssigned = CompAssignableToPawn_PawnStorage.compAssiblables.FirstOrDefault(c => c.assignedPawns.Contains(prisoner))?.parent;
+        if (existingAssigned != null)
+        {
+            return existingAssigned;
+        }
 
-		if (CompAssignableToPawn_PawnStorage.compAssiblables.FirstOrDefault(c => c.HasFreeSlot && c.OwnerType == BedOwnerType.Prisoner) is not { } assignable) return null;
-		if (assign) assignable.TryAssignPawn(prisoner);
-		return assignable.parent;
-	}
+        if (CompAssignableToPawn_PawnStorage.compAssiblables.FirstOrDefault(c => c.HasFreeSlot && c.OwnerType == BedOwnerType.Prisoner) is not { } assignable) return null;
+        if (assign) assignable.TryAssignPawn(prisoner);
+        return assignable.parent;
+    }
+
+    public static ThingWithComps GetStorageEntityOrAnimal(Pawn prisoner, bool assign = false)
+    {
+        ThingWithComps existingAssigned = CompAssignableToPawn_PawnStorage.compAssiblables.FirstOrDefault(c => c.assignedPawns.Contains(prisoner))?.parent;
+        if (existingAssigned != null)
+        {
+            return existingAssigned;
+        }
+
+        var bedOwnerType = BedOwnerType.Prisoner;
+
+        if (prisoner.Faction == Faction.OfPlayer) bedOwnerType = BedOwnerType.Colonist;
+
+        if (CompAssignableToPawn_PawnStorage.compAssiblables.FirstOrDefault(c => c.HasFreeSlot && c.OwnerType == bedOwnerType) is not { } assignable) return null;
+        if (assign) assignable.TryAssignPawn(prisoner);
```

Comment:
I assume this might be the cause of your issue, can you even assign entities?

Reply:
Yup, they're assignable. All working now.